api/survey.ts

```diff
@@ -0,0 +1,59 @@
+import handler from './handler';
+import db from './db';
+
+export default handler(async (request : any) => {
+
+  let connection = db();
+  const client = await connection.connect()
+
+  // Take data from request and build sql call from it
+  // TODO: Move this data structure to it's own class
+  const {
+    //Sharded Values
+    patientid,
+    completiondate,
+    completiontime,
+    //TFI Survey
+    tfisurveyid,
+    tfi_i,
+    tfi_sc,
+    tfi_c,
+    tfi_si,
+    tfi_a,
+    tfi_r,
+    tfi_q,
+    tfi_e,
+    tfi_overallscore,
+    //THS Survey
+    thssurveyid,
+    ths_sectiona,
+    ths_sectionb,
+    ths_sectionc,
+    //TS Survey
+    tssurveyid,
+    ts_type
+  } = request.body
+
+  var tfisurvey_sql = "INSERT INTO tfisurvey (tfisurveyid, patientid, completiondate, completiontime, tfi_i, tfi_sc, tfi_c, tfi_si, tfi_a, tfi_r, tfi_q, tfi_e, tfi_overallscore) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13)"
+  var tfisurvey_values = [tfisurveyid, patientid, completiondate, completiontime, tfi_i, tfi_sc, tfi_c, tfi_si, tfi_a, tfi_r, tfi_q, tfi_e, tfi_overallscore]
+  
+  var thssurvey_sql = "INSERT INTO thssurvey (thssurveyid, patientid, completiondate, completiontime, ths_sectiona, ths_sectionb, ths_sectionc) VALUES ($1, $2, $3, $4, $5, $6, $7)"
+  var thssurvey_values = [thssurveyid, patientid, completiondate, completiontime, ths_sectiona, ths_sectionb, ths_sectionc]
+  
+  var tssurvey_sql = "INSERT INTO tssurvey (tssurveyid, patientid, completiondate, completiontime, ts_type) VALUES ($1, $2, $3, $4, $5)"
+  var tssurvey_values = [tssurveyid, patientid, completiondate, completiontime, ts_type]
+
+  try {
+    await client.query('BEGIN')
+    await connection.query(tfisurvey_sql, tfisurvey_values)
```

Comment:
I think you can remove every `await` except the last one because you don't need to wait for one query to finish before sending the next one.

Reply:
Actually, use `await Promise.all(conn..., conn... ...)` to simultaneously wait for the three survey submit promises, rather than waiting for one at a time.